lib/Pipeline/InspectableLLVMContainer.cpp

```diff
@@ -0,0 +1,26 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/DenseSet.h"
+#include "llvm/IR/GlobalValue.h"
+
+#include "revng/Pipeline/InspectableLLVMContainer.h"
+#include "revng/Pipeline/LLVMContainer.h"
+#include "revng/Pipeline/PureLLVMPipe.h"
+#include "revng/Pipeline/Target.h"
+
+using namespace Pipeline;
+using namespace llvm;
+
+char LLVMContainer::ID = '0';
+
+void PureLLVMPipe::run(LLVMContainer &Container) {
+  legacy::PassManager Manager;
+  auto *Registry = PassRegistry::getPassRegistry();
+  for (const auto &Element : PassNames) {
+
+    Manager.add(Registry->getPassInfo(Element)->createPass());
```

Comment:
Nice, didn't know you could get passes by names like this!

Reply:
i struggled a lot to find this